File: bms/go/triangolatte/Point_Scale/Point_Scale.go
package test 

import (
    "math"
)

var (
)

const (
)

func (p Point) Scale(f float64) Point {
	norm := float64(math.Sqrt(float64(p.X*p.X + p.Y*p.Y)))
	return Point{p.X / norm * f, p.Y / norm * f}
}

type Point struct {
	X, Y float64
}

File: translations_only/fail/go/triangolatte/Point_Scale/claude3/BaseRepair/Point_Scale.rs

use std::f64::consts::PI;

struct Point {
    x: f64,
    y: f64,
}

fn scale(p: Point, f: f64) -> Point {
    let norm = (p.x * p.x + p.y * p.y).sqrt();
    Point {
        x: p.x / norm * f,
        y: p.y / norm * f,
    }
}
